Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/aws_utils.py

```diff
@@ -103,7 +103,15 @@ def list_s3_buckets(region: str = "") -> List[str]:
     bucket_names: List[str] = []
     bucket: Dict[str, any]
     for bucket in response["Buckets"]:
-        bucket_names.append(bucket["Name"])
+        try:
+            bucket_name: str = bucket["Name"]
+            location_response: Dict[str, any] = s3_client.get_bucket_location(Bucket=bucket_name)
+            if ((location_response["LocationConstraint"] == region) or
+                    (not location_response["LocationConstraint"] and region == "us-east-1")):
```

Comment:
nit: I would add a comment and link here as to why we default to us-east-1 to make code easier for future maintainers.

Reply:
Yeah, it's a little bit confusing here. Is us-east-1 the default region for s3 bucket?

Reply:
> Yeah, it's a little bit confusing here. Is us-east-1 the default region for s3 bucket?

See PR description for some of the why (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.get_bucket_location). Unsure if all buckets in us-east-1 have no location information, or just legacy buckets; us-east-1 is the default region for an S3 bucket so its possible location was/is not set for buckets in region.

Reply:
It is just a special check case for IAD bucket.
I guess it is a legacy issue of S3 service, feels like S3 team didn't backfill s3 bucket location data, so they keep it null forever. (But boto3 should be easy to adopt the issue by filling this field instead of returning the data directly)